### PR TITLE
more cleanups

### PR DESCRIPTION
**Pull request by matthiaskrgr**
* use starts_with() instead of chars().next() == Some(x)
* use subsec_micros() instead of subsec_nanos() / 1000
* use for (idx, item) in iter.enumerate() instead of manually counting loop iterations with variables
* use values() or keys() respectively when iterating only over keys or values of maps.

### CODE REVIEW

File: src/librustc/ty/layout.rs [hunk 1]
```diff
@@ -1382,10 +1382,8 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
 
         // Write down the order of our locals that will be promoted to the prefix.
         {
-            let mut idx = 0u32;
-            for local in ineligible_locals.iter() {
-                assignments[local] = Ineligible(Some(idx));
-                idx += 1;
+            for (idx, local) in ineligible_locals.iter().enumerate() {
+                assignments[local] = Ineligible(Some(idx as u32));
```
Comment (eddyb) [inline src/librustc/ty/layout.rs]: cc @tmandry @Zoxc

File: src/librustc_resolve/late/lifetimes.rs [hunk 2]
```diff
@@ -347,9 +347,7 @@ fn krate(tcx: TyCtxt<'_>) -> NamedRegionMap {
             lifetime_uses: &mut Default::default(),
             missing_named_lifetime_spots: vec![],
         };
-        for (_, item) in &krate.items {
-            visitor.visit_item(item);
-        }
+        krate.items.values().for_each(|item| visitor.visit_item(item))
```
Comment (eddyb) [inline src/librustc_resolve/late/lifetimes.rs]: I prefer avoiding `for_each` as an anti-pattern, tbh, but the `.values()` change seems fine.

File: src/libstd/sys/unix/net.rs [hunk 3]
```diff
@@ -280,7 +280,7 @@ impl Socket {
                 };
                 let mut timeout = libc::timeval {
                     tv_sec: secs,
-                    tv_usec: (dur.subsec_nanos() / 1000) as libc::suseconds_t,
+                    tv_usec: dur.subsec_micros() as libc::suseconds_t,
```
Comment (eddyb) [inline src/libstd/sys/unix/net.rs]: cc @alexcrichton

**Review by eddyb — APPROVED** [review]
r=me modulo `for_each` nit (I don't expect the cc's to require any changes)

File: src/librustc_resolve/late/lifetimes.rs (hunk 2 above)
Comment (matthiaskrgr) [reply src/librustc_resolve/late/lifetimes.rs]: sure, I'll change it

**Review by eddyb — APPROVED** [review]

**Pull request merged**